list-agreements.lua

```diff
@@ -1,120 +1,219 @@
-        return "temple"
-    elseif get_location_type(agr) == df.abstract_building_type.TEMPLE and get_location_tier(agr) == 2 then
-        return "temple complex"
-    elseif get_location_type(agr) == df.abstract_building_type.GUILDHALL and get_location_tier(agr) == 1 then
-        return "guildhall"
-    elseif get_location_type(agr) == df.abstract_building_type.GUILDHALL and get_location_tier(agr) == 2 then
-        return "grand guildhall"
-    end
+function get_petition_date(agr)
+    local agr_year = agr.details[0].year
+    local agr_year_tick = agr.details[0].year_tick
+    local julian_day = math.floor(agr_year_tick / 1200) + 1
+    local agr_month = math.floor(julian_day / 28) + 1
+    local agr_day = julian_day % 28
+    return string.format("%03d-%02d-%02d",agr_year, agr_month, agr_day)
 end
 
-function get_petition_date(agr)
-    agr_year = agr.details[0].year
-    agr_year_tick = agr.details[0].year_tick
-    julian_day = math.floor(agr_year_tick / 1200) + 1
-    agr_month = math.floor(julian_day / 28) + 1
-    agr_day = julian_day % 28
-    return agr_year.."-"..agr_month.."-"..agr_day
+function get_petition_age(agr)
+    local agr_year_tick = agr.details[0].year_tick
+    local agr_year = agr.details[0].year
+    local cur_year_tick = df.global.cur_year_tick
+    local cur_year = df.global.cur_year
+    local del_year, del_year_tick
+    --delta, check to prevent off by 1 error, not validated
+    if cur_year_tick > agr_year_tick then
+        del_year = cur_year - agr_year
+        del_year_tick = cur_year_tick - agr_year_tick
+    else
+        del_year = cur_year - agr_year - 1
+        del_year_tick = agr_year_tick - cur_year_tick
+    end
+    local julian_day = math.floor(del_year_tick / 1200) + 1
+    local del_month = math.floor(julian_day / 28)
+    local del_day = julian_day % 28
+    return {del_year,del_month,del_day}
 end
 
 function get_guildhall_profession(agr)
-    prof = agr.details[0].data.Location.profession
+    local prof = agr.details[0].data.Location.profession
+    local profname = 'profession-less'
     profname = string.lower(df.profession[prof])
+    -- *VERY* important code follows
+    if string.find(profname, "man") then
+        profname = string.gsub(profname,"man",string.lower(dfhack.units.getRaceNameById(df.global.ui.race_id)))
+    end
     return profname:gsub("_", " ")
 end
 
-function get_religion_name(agr)
-    religion_id = agr.details[0].data.Location.deity_data.Religion
-    religion_name = dfhack.TranslateName(df.global.world.entities.all[religion_id].name, true)
-    return religion_name
+function get_agr_party_name(agr)
+    --assume party 0 is guild/order, 1 is local government as siteid = playerfortid
+    local party_id = agr.parties[0].entity_ids[0]
+    local party_name = 'An Unknown Entity or Group'
+    party_name = dfhack.TranslateName(df.global.world.entities.all[party_id].name, true)
+    return party_name
```

Comment:
not sure this does what i want it to: return the default if dfhack.TranslateName fails

flip them and `if not party_name then <default>` ?

Reply:
a common idiom in Lua is `x = a or b` to mean
```
x = a
if not x then
    x = b
end
```

so in this case you could write `return dfhack.TranslateName() or 'An Unknown Entity or Group'` (assuming `TranslateName` returns nil or false on failure).